### PR TITLE
CLOUDP-159640: kubernetes config apply command

### PR DESCRIPTION
**Pull request by helderjs**
## Proposed changes

This PR adds a new (sub) command which exports Atlas resources configurations and applies to a running Kubernetes cluster.
Note: A new ubuntu based build variant was added to configure Kubernetes tooling for e2e tests

_Jira ticket:_ CLOUDP-159640

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/kubernetes/operator/config_apply.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/features"
+	akov1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const numberOfExistingResources = 7
```
Comment (igor-karpukhin) [inline internal/kubernetes/operator/config_apply.go:28]: Maybe we can calculate this by checking the length of the resources list `resourcesToVersionMap[latestOperatorVersion]`?

**Review by igor-karpukhin — APPROVED** [review]
@helderjs great job! 👍

File: build/ci/setup_operator.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Copyright 2021 MongoDB Inc
```
Comment (gssbzn) [inline build/ci/setup_operator.sh:3]:
```suggestion
# Copyright 2023 MongoDB Inc
```

File: build/ci/setup_operator.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  checkCmd="while ! ${workdir}/bin/kubectl --namespace mongodb-atlas-system get pods --selector app.kubernetes.io/instance=mongodb-atlas-kubernetes-operator --output jsonpath={.items[0].status.phase} 2>/dev/null | grep -q Running ; do printf .; sleep 1; done"
+  timeout --foreground 1m bash -c "${checkCmd}" || true
+  if [ "$("$workdir"/bin/kubectl --namespace mongodb-atlas-system get pods --selector app.kubernetes.io/instance=mongodb-atlas-kubernetes-operator --output jsonpath='{.items[0].status.phase}' | grep -q Running)" = 0 ]; then
+      echo "Operator hasn't reached RUNNING state after 1 minute. The full yaml configuration for the pod is:"
+      kubectl --namespace mongodb-atlas-system get pods --selector app.kubernetes.io/instance=mongodb-atlas-kubernetes-operator --output yaml#
+
+      echo "Operator failed to start, exiting"
+      exit 1
+  fi
+fi
```
Comment (gssbzn) [inline build/ci/setup_operator.sh:78]: missing new line

File: internal/cli/atlas/kubernetes/config/apply.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+		return fmt.Errorf("unable to auto detect params: %w", err)
+	}
+
+	if opts.targetNamespace == "" {
+		opts.targetNamespace = operatorDeployment.Namespace
+	}
+
+	if opts.operatorVersion == "" {
+		if !strings.HasPrefix(operatorDeployment.Spec.Template.Spec.Containers[0].Image, fmt.Sprintf("%s:", containerImage)) {
+			return fmt.Errorf("unable to auto detect operator version. you should explicitly set operator version if you are running an openshift certified installation")
```
Comment (gssbzn) [inline internal/cli/atlas/kubernetes/config/apply.go:83]:
```suggestion
			return errors.Newf("unable to auto detect operator version. you should explicitly set operator version if you are running an openshift certified installation")
```

File: internal/cli/atlas/kubernetes/config/apply.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+		if len(deployments.Items) > 0 {
+			if err != nil {
+				return nil, err
+			}
+
+			return &deployments.Items[0], nil
+		}
+	}
+
+	return nil, fmt.Errorf("couldn't to find operator installed in any accessible namespace")
```
Comment (gssbzn) [inline internal/cli/atlas/kubernetes/config/apply.go:258]:
```suggestion
	return nil, errors.New("couldn't to find operator installed in any accessible namespace")
```

File: internal/cli/atlas/kubernetes/config/apply_test.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 MongoDB Inc
```
Comment (gssbzn) [inline internal/cli/atlas/kubernetes/config/apply_test.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: internal/kubernetes/operator/config_apply.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+		return err
+	}
+
+	sortedResources := sortResources(ProjectResources, DeploymentResources)
+
+	for _, objects := range sortedResources {
+		for _, object := range objects {
+			ctrlObj, ok := object.(client.Object)
+			if !ok {
+				return fmt.Errorf("unable to apply resource")
```
Comment (gssbzn) [inline internal/kubernetes/operator/config_apply.go:98]:
```suggestion
				return errors.New("unable to apply resource")
```

File: test/e2e/atlas/operator_helper_test.go [hunk 8]
```diff
@@ -0,0 +1,203 @@
+// Copyright 2021 MongoDB Inc
```
Comment (gssbzn) [inline test/e2e/atlas/operator_helper_test.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**